webpack/scenes/ContentViews/Details/Filters/CVModuleStreamFilterContent.js

```diff
@@ -0,0 +1,224 @@
+import React, { useState, useCallback } from 'react';
+import useDeepCompareEffect from 'use-deep-compare-effect';
+import PropTypes from 'prop-types';
+import { shallowEqual, useSelector, useDispatch } from 'react-redux';
+import { TableVariant } from '@patternfly/react-table';
+import { Tabs, Tab, TabTitleText, Split, SplitItem, Button, Dropdown, DropdownItem, KebabToggle } from '@patternfly/react-core';
+import { STATUS } from 'foremanReact/constants';
+import { translate as __ } from 'foremanReact/common/I18n';
+
+import onSelect from '../../../../components/Table/helpers';
+import TableWrapper from '../../../../components/Table/TableWrapper';
+import {
+  selectCVFilterModuleStream,
+  selectCVFilterModuleStreamStatus,
+  selectCVFilterModuleStreamError,
+  selectCVFilters, selectCVFilterDetails,
+} from '../ContentViewDetailSelectors';
+import getContentViewDetails, {
+  addCVFilterRule, removeCVFilterRule, getCVFilterModuleStreams,
+  deleteContentViewFilterRules, addContentViewFilterRules,
+} from '../ContentViewDetailActions';
+import AddedStatusLabel from '../../../../components/AddedStatusLabel';
+import AffectedRepositoryTable from './AffectedRepositories/AffectedRepositoryTable';
+
+const CVModuleStreamFilterContent = ({
+  cvId, filterId, showAffectedRepos, setShowAffectedRepos,
+}) => {
+  const dispatch = useDispatch();
+  const { results: filterResults } =
+    useSelector(state => selectCVFilters(state, cvId), shallowEqual);
+  const response = useSelector(state =>
+    selectCVFilterModuleStream(state, cvId, filterId), shallowEqual);
+  const status = useSelector(state =>
+    selectCVFilterModuleStreamStatus(state, cvId, filterId), shallowEqual);
+  const error = useSelector(state =>
+    selectCVFilterModuleStreamError(state, cvId, filterId), shallowEqual);
+  const filterDetails = useSelector(state =>
+    selectCVFilterDetails(state, cvId, filterId), shallowEqual);
+  const { repositories = [] } = filterDetails;
+  const [rows, setRows] = useState([]);
+  const [metadata, setMetadata] = useState({});
+  const [searchQuery, updateSearchQuery] = useState('');
+  const [activeTabKey, setActiveTabKey] = useState(0);
+  const loading = status === STATUS.PENDING;
+  const [bulkActionOpen, setBulkActionOpen] = useState(false);
+  const deselectAll = () => setRows(rows.map(row => ({ ...row, selected: false })));
+  const toggleBulkAction = () => setBulkActionOpen(prevState => !prevState);
+  const hasAddedSelected = rows.some(({ selected, added }) => selected && added);
+  const hasNotAddedSelected = rows.some(({ selected, added }) => selected && !added);
+  const columnHeaders = [
+    __('Name'),
+    __('Stream'),
+    __('Version'),
+    __('Context'),
+    __('Status'),
+  ];
+
+  const buildRows = useCallback((results) => {
+    const newRows = [];
+    const filterRules = filterResults.find(({ id }) => id === Number(filterId))?.rules || [];
```

Comment:
Try `filterResults?.find` 

Sometimes refreshing the page fails on this. Even better would be to use the useSelector(state => selectCVFiltersStatus(state, cvId)); and then check if the status is not pending.

Something like this: https://github.com/Katello/katello/blob/31ea5f8cf3364dcced2f899d1e0aa30fbacbe845/webpack/scenes/ContentViews/Details/ComponentContentViews/ContentViewComponents.js#L188